novb12-sp2

bnaru opened the pull request:
Link to application: https://eso.vse.cz/~novb12/sp2/

bnaru committed:
novb12-sp2

Link to application: https://eso.vse.cz/~novb12/sp2/

bnaru committed:
Verze 2 (finální k obhajobě)

Změna či přidání většiny věcí

bnaru committed:
Verze 2.1 (finální k opravě)

Patří k verzi 2, kde nebyly nahrány obrázky po kompresi.

bnaru committed:
Stará verze favicon

bnaru committed:
Stará verze favicon

bnaru committed:
Stará verze favicon

bnaru committed:
Stará verze favicon

bnaru committed:
Staré logo

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+// NAČÍTÁNÍ Z LOCAL STORAGE
+function loadWatchlist() {
+  document.getElementById('movie-list').innerHTML = localStorage.getItem('mySavedMovies');
```
do localStorage ukladame pouze cista data, ne cely HTML

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+
+  //OMDb API -> http://www.omdbapi.com/ (odkud čerpám data o filmech)
+  const apikey = '6ffee1f0' //unikátní API key
+  const url = 'http://www.omdbapi.com/?apikey=' + apikey //URL přes kterou lze provádět vyhledávání
```
pouzijte string interpolation

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+  const apikey = '6ffee1f0' //unikátní API key
+  const url = 'http://www.omdbapi.com/?apikey=' + apikey //URL přes kterou lze provádět vyhledávání
+
+  const movieForm = $('#form'); //celý formulář vyplňovaný uživatelem
```
proc tady vybirate element pomoci jQuery, jinde pres `document.querySelector` a nekde zase pomoci `document.getElementById`?

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+  const movieInput = $('#movie-input'); //název filmu, který uživatel píše do okna a potvrzuje ho tlačítkem
+
+  // POZDRAV PRO UŽIVATELE (závislý na denní době)
+  var sentence = ', I hope you have something to watch and you will enjoy your free time! If you want to have some inspiration click on the watch guide below...';
```
misto var pouzivejte const nebo let

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+  var sentence = ', I hope you have something to watch and you will enjoy your free time! If you want to have some inspiration click on the watch guide below...';
+  var time = new Date().getHours();
+  if (time < 12) {
+    greeting = 'Good morning' + sentence;
```
chybi deklarace promenne

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+
+    $.ajax({
+      method: 'GET',
+      url: url + '&t=' + movie,
```
tady pouzijte vyhledavani `by search`  podle dokumentace je tam parametr `s` a ne `t`

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+        //console.log(data); -> není třeba to zobrazovat na konzoli (důležité info je vykresleno do webu)
+
+        //Pokud data existují v API (tzn. nemají hodnotu undefined)
+        if (data.Title != undefined) {
```
`if (data.Title)`

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+          if (document.getElementById('added-' + data.Title) === null) {
+            var li = document.createElement('li'); //každý film má vytvořený svůj vlastní div s informacemi
+            li.setAttribute('class', 'must-see-movie'); //div.setAttribute(' ',' ');
+            li.setAttribute('id', 'added-' + data.Title);
```
tady byste mela pouzit `data.imdbID` a ne `data.Title` kdyz uz

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+  movieForm.submit(function (e) {
+    e.preventDefault();
+
+    const movie = movieInput.val().trim(); //.trim() zbaví input prázdných míst - jediné omezení u vstupu (víceslovné filmy by se při několikanásobném vstupu nepovedly -> nepočítáme tedy s ním)
```
a co kdyz uzivatel tam zada pouze nekolik mezer?

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+
+            var img = document.createElement('img'); //Div s plakátem filmu/seriálu
+            img.src = data.Poster;
+            img.alt = 'Movie poster';
```
alt bude 
```js
`${data.Title} Poster`
```

nvbach91 commented on www/novb12/sp2/assets/js/main.js:
```diff
@@ -0,0 +1,169 @@
+        if (data.Title != undefined) {
+
+          //Pokud už není v seznamu, tak se provede přidání do seznamu
+          if (document.getElementById('added-' + data.Title) === null) {
```
hacker 101, spravneji byste mela mit promennou s objektem ve kterem drzite tyhle oblibene filmy, kde klicem budou id techto filmu

nvbach91 reviewed (changes requested):
- veskery kod dejte do closure `$(document).ready(() => { ... })`
- zapracujte endpoint search, pricemz se zobrazi docasny seznam filmu, mezi ktere provedu vyber pro pridani do oblibenych, zobrazeni oblibenych filmu bude v samostatne sekci (prepinani mezi mody vyhledavani a zobrazeni oblibenych)

bnaru committed:
Verze s úkoly z obhajoby

Oprava většiny věcí kromě local storage a přidávání, kód je však z části zachován zakomentovaný.

bnaru committed:
Stejné jako předchozí verze (pouze zformátování)

Zformátování css a html - zbytek byl v pořádku
